homeassistant/components/telegram_bot/__init__.py

```diff
@@ -876,7 +876,7 @@ def _get_message_data(self, msg_data):
 
         if (
             msg_data["from"].get("id") not in self.allowed_chat_ids
-            and msg_data["message"]["chat"].get("id") not in self.allowed_chat_ids
+            and msg_data["chat"].get("id") not in self.allowed_chat_ids
```

Comment:
See this comment, why it's not as simple as this:
https://github.com/home-assistant/core/pull/60060#discussion_r754337150